[ADD] Payroll: New Salary doc (#14806)
odoo/documentation: content/applications/hr/payroll/salaries.rst

========
Salaries
========

In Odoo, salaries are calculated and determined by five factors: salary :ref:`structure types
<payroll/structure-types>`, salary :ref:`structures <payroll/structures>`, :ref:`rules
<payroll/rules>`, :ref:`rule parameters <payroll/rule-parameters>`, and :ref:`other input types
<payroll/other-input>`. All of these inputs work together to calculate every employee's pay.

Every salary structure *type* contains one or more salary *structure* within it. Each *structure*
contains salary *rules*, and every *rule* has rule *parameters* that it follows. In addition, there
are other inputs which may need to be added and calculated to determine a salary.

> When payslips are calculated, Odoo determines how much time the employee worked from their

Reviewer: ```suggestion
When payslips are calculated, Odoo calculates the employee’s worked time from their
```

:doc:`work entries <work_entries>`, and then calculates their pay using the various rules and rule
parameters for the employee's associated salary structure type.

.. _payroll/structure-types:

Structure types
===============

In Odoo, structure *types* are used to group related structures together. When contracts specify
a structure type, **only** structures within that structure type can be used to calculate pay. Each
structure type houses individual structures within them, each containing a set of rules for
processing a timesheet entry.

Structure types define how often an employee gets paid, their working hours, its default salary
structure, and if wages are based on a salary (fixed) or how many hours the employee worked
(varied).

.. example::
   A structure type called `Employee` contains two different structures within it: a `Regular Pay`
   structure which includes all the separate rules for processing regular pay, and an `End of Year
   Bonus` structure, which includes the rules **only** for the end of year bonus. Both the `Regular
   Pay` structure and `End of Year Bonus` structure are structures *within* the `Employee` structure

>    type.

Reviewer: ```suggestion
   Bonus` structure, which includes the rules **only** for the end of year bonus. Both belong to the same `Employee` structure
   type.
```

The different *structure types* can be seen by navigating to :menuselection:`Payroll app -->

> Configuration --> Structure Types`.

Reviewer: ```suggestion
View existing *structure types* by navigating to :menuselection:`Payroll app --> Configuration -->
Structure Types`.
```

Two default structure types are preconfigured in Odoo: :guilabel:`Employee` and :guilabel:`Worker`.

Typically, :guilabel:`Employee` is used for salaried employees, which is why the :guilabel:`Default
Wage Type` is a :guilabel:`Fixed Wage`, and :guilabel:`Worker` is typically used for employees paid
by the hour, so the wage type is :guilabel:`Hourly Wage`.

.. note::
   If using a country-specific :doc:`payroll localizations <payroll_localizations>`, it is
   recommended to use the structure in the corresponding country localization document.

.. image:: salaries/structure-type.png
   :alt: List of all currently configured structure types available to use.

.. _payroll/new-structure-type:

New structure type
------------------

If one of the default structure types is not suited for the needs of the company, a new structure
type can be created. To make a new structure type, click the :guilabel:`New` button, and a blank

> structure type form appears.

Author: That's too funny- I feel like I over-navigate/over-explain, so originally I had it in here, but then took it out since I was worried I was "repeating myself" too much! =D

  from the drop-down menu. All available working hours for the currently selected company appear in
  the drop-down menu. The default working hours that are preconfigured in Odoo is the
  :guilabel:`Standard 40 hours/week` option. If the needed working hours do not appear in the list,
  a :ref:`new set of default working hours can be created <new-default-working-hours>`.
- :guilabel:`Regular Pay Structure`: Type in the name for the regular pay structure. It is used as
  the default option when generating payslips.
- :guilabel:`Default Work Entry Type`: Select the default work entry type which is used to create
  all work entries for the employee.

.. image:: salaries/new-structure.png
   :alt: New structure type form to fill out when creating a new structure type.

.. _new-default-working-hours:

New default working hours
-------------------------

To make new default working hours, type the name for the new working hours in the :guilabel:`Default
Working Hours` field on the new structure type form. Click :guilabel:`Create and edit`. A default
working hours form pops up. The default working hours form has two sections, a general information

> section and a tab listing out all the individual working hours by day and time. When the form is

Reviewer: ```suggestion
working hours form pops up. The default working hours form has two sections: a general information
section, and a tab listing all the individual working hours by day and time. When the form is
```

completed, click :guilabel:`Save & Close`.

- :guilabel:`Name`: Type in the name for the new default working hours. This should be descriptive
  and clear to understand, such as `Standard 20 Hours/Week`.
- :guilabel:`Flexible Hours`: Tick the checkbox if employees can work hours different from the
  default company work hours.
- :guilabel:`Company Full Time`: Enter the number of hours per week an employee needs to work to be
  considered a full-time employee. Typically, this is approximately 40 hours, and this number
  affects what types of benefits an employee can receive, based on their employment status
  (full-time vs part-time).
- :guilabel:`Average Hour Per Day`: The average hours per day field is auto-populated, based on the
  working hours configured in the :guilabel:`Working Hours` tab. This entry affects resource
  planning, since the average daily hours affect what resources can be used, and in what quantity,

>   per workday.

Reviewer: ```suggestion
- :guilabel:`Average Hour Per Day`: Auto-populated based on the working hours in the
  :guilabel:`Working Hours` tab. This entry affects resource
  planning by determining how many resources can be used per workday.
```

  configured, using a 24-hour time format.

  To adjust any of these hours, click on the specific field to change, and make the adjustment using
  the drop-down menus, or in the specific case of the times, type in the desired time.

  .. note::
     Keep in mind, working hours are company-specific, and cannot be shared between companies. Each
     company needs to have their own working hours set.

  .. tip::
     If the working hours are not consistent each week, and the hours are on a bi-weekly schedule
     instead, click the :guilabel:`Switch to 2 week calendar` button at the top of the new default
     working hours form. This changes the :guilabel:`Working Hours` tab to display two weeks of
     working times that can be adjusted.

.. _payroll/structures:

Structures
==========

*Salary structures* are the different situations in which an employee could be paid within a
specific *structure*, and are specifically defined by various rules.

The amount of structures a company needs for each structure type depends on how many different ways
employees are paid, and how their pay is calculated. For example, a common structure that could be

> useful to add may be a `Bonus`.

Reviewer: ```suggestion
employees are paid, and how their pay is calculated. A common example of an additional
structure is a `Bonus`.
```

To view all the various structures for each structure type, go to :menuselection:`Payroll app -->
Configuration --> Structures`.

Each :ref:`structure type <payroll/structure-types>` lists the various structures associated with
it. Each structure contains a set of rules that define it.

.. image:: salaries/salary-structure.png
   :alt: All available salary structures.

Click on a structure to view its :guilabel:`Salary Rules`. These rules define how the payslip will
be computed for the employee.

.. note::
   After installing a :doc:`payroll localization <payroll_localizations>`, relevant structures are
   installed and appear in this list.

.. image:: salaries/structure-regular-pay-rules.png
   :alt: Salary structure details for Regular Pay, listing all the specific Salary Rules.

.. _payroll/rules:

Rules

> =====

Reviewer: For the first time, I recommend choosing the longer title, because in the intro and in the structure section above, you mention the concept "salary rules". To make sure the reader who wants to learn about salary rules comes to the rules section, what do you think about naming this super on-the-nose: "Salary rules"?

```suggestion
Salary rules
============
```

  is required.

Computation
~~~~~~~~~~~

- :guilabel:`Amount Type`: Select from the drop-down menu whether the amount is a :guilabel:`Fixed
  Amount`, a :guilabel:`Percentage (%)`, :guilabel:`Other Input`, or a :guilabel:`Python Code`.
  Depending on what is selected, the fixed amount, percentage, other input, or Python code needs to
  be entered next. This field is required.

Company contribution
~~~~~~~~~~~~~~~~~~~~

- :guilabel:`Partner`: If another company financially contributes to this rule, select the company
  from the drop-down menu.

Description tab
---------------

Provide any additional information in this tab to help clarify the rule. This tab only appears in
the rule form.

Accounting tab
--------------

- :guilabel:`Debit Account`: Select the debit account from the drop-down menu the rule affects.
- :guilabel:`Credit Account`: Select the credit account from the drop-down menu the rule affects.
- :guilabel:`Split account line based on name`: Tick the checkbox to split the accounting entry
  according to the payslip line name.
- :guilabel:`Not computed in net accountability`: If checked, the amount of the rule is shown

>   independently from the Net Salary, to allow for better reporting in the **Accounting** app.

Reviewer: Is Net Salary a field? If so, let's italicize it! If not, let's remove the capitalization and simply say "net salary"
